homeassistant/components/epic_games_store/config_flow.py

```diff
@@ -0,0 +1,102 @@
+"""Config flow for Epic Games Store integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from epicstore_api import EpicGamesStoreAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import CONF_LOCALE, CONF_SUPPORTED_LOCALES, DOMAIN
+from .coordinator import not_handle_service_errors
+from .helper import get_country_from_locale
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_LOCALE): vol.In(CONF_SUPPORTED_LOCALES),
+    }
+)
+
+
+def get_default_locale(hass: HomeAssistant) -> str:
+    """Get default locale code based on Home Assistant config."""
+    locale_code = f"{hass.config.language}-{hass.config.country}"
+    if locale_code in CONF_SUPPORTED_LOCALES:
+        return locale_code
+    if hass.config.language in CONF_SUPPORTED_LOCALES:
+        return hass.config.language
+    return "en-US"
+
+
+async def validate_input(
+    hass: HomeAssistant, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+    api = EpicGamesStoreAPI(
+        user_input[CONF_LOCALE], get_country_from_locale(user_input[CONF_LOCALE])
+    )
+    # pylint: disable-next=protected-access
+    api._get_errors = not_handle_service_errors
```

Comment:
Why do we do this?

Reply:
The library is handling errors from the API by raising an exception, like so, we are not getting the data !
EVEN IF the data is well retreived but the error attr is present telling a minor issue with 1 or 2 items. That little issue could be worked around, or correspond to a data we are not even using ... 😒 this is the case of the error 1004 that you can found in the fixtures older.

Having this error attr do not block the usage of the API as we are having all the data needed !

`api._get_errors` is a function called AFTER the API fetch, handling some useless functional errors, other HTTP/Network errors are raise before and are not blocked.